### PR TITLE
Refine StreamLoadPipe::append_and_flush()

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

## Summary
Should flush the data in `_write_buf` before writing the data passed by `append_and_flush`

### CODE REVIEW

File: be/src/runtime/stream_load/stream_load_pipe.h [hunk 1]
```diff
@@ -200,41 +204,8 @@ class StreamLoadPipe : public MessageBodySink, public FileReader {
     }
 
 private:
-    // read the next buffer from _buf_queue
-    Status _read_next_buffer(std::unique_ptr<uint8_t[]>* out, size_t* out_cap, size_t* out_sz, size_t padding) {
```
Comment (rickif) [inline be/src/runtime/stream_load/stream_load_pipe.h:204]: Nice catch. I'm about to delete this.

**Comment by rickif** [comment]
Could `Status append(const ByteBufferPtr& buf)` also be removed?

**Comment by sduzh** [comment]
> Could `Status append(const ByteBufferPtr& buf)` also be removed?

The unit test still relies on this method now.

**Comment by sduzh** [comment]
> > Could `Status append(const ByteBufferPtr& buf)` also be removed?
> 
> The unit test still relies on this method now.

Removed, PTAL

**Review by rickif — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**